Automattic/jetpack
# Jetpack Sync : update should_initialize_sender_enqueue to honor filtered value vs overwriting.

Currently when a new sync action is enqueued we add a filter that will always enable sending of data on shutdown except when cron is disabled. 

This PR ensures we don't setup the shutdown sender if it has been determined sending is disabled by previous filters.

#### Changes proposed in this Pull Request:
* Update should_initialize_sender_enqueue to use the $enabled property it is filtering.

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:

* checkout branch
* Create a Post
* Verify action is queued and sent (can put log statement in send_data or view kibana logs)
* Add this filter to your site
```add_filter( 'jetpack_sync_sender_should_load', '__return_false', 1, 0 );```
* Create a Post
* Verify action is in the queue
  `$listener = \Automattic\Jetpack\Sync\Listener::get_instance();`
 `$queue = $listener->get_sync_queue();`
 `$queue->size();`

#### Proposed changelog entry for your changes:
* Jetpack Sync : update should_initialize_sender_enqueue to honor filtered value vs overwriting.
